client/app/components/packages/pas-form/edit.hbs

```diff
@@ -606,6 +613,7 @@
               <saveable-form.radio
                 @value={{radio.code}}
                 @groupValue={{saveable-form.saveableChanges.dcpDiscressionaryfundingforffordablehousing}}
+                @changed={{fn this.updateAttr saveable-form.saveableChanges "dcpHousingunittype" ""}}
```

Comment:
why not use `mut`? i really want to avoid this style of using string references to key names.

Reply:
have you considered other options, too? why is this the best path? what about:

making a new component
changing the existing component
using built-in helpers for managing state

do you think this approach will harm future maintainability?

Reply:
@allthesignals I updated the radio buttons in `edit.hbs` to use the `mut` helper thanks for the comment!

As for creating editing existing components (either the `radio-button` or `saveable-form` components), after looking over it a bit I don't think that making that change would simplify this implementation. We would still be required to pass in extra parameters (e.g. `@attributeToReset`) which might complicate the rendering of the radio button -- by either changing what is essentially source code in the `radio-button` component or needing to pass in an extra parameter when we render the `saveable-form.radio` component. The way that the `saveable-form.radio` button is rendered now in `edit.hbs`, there is no input attribute passed in, so that would have to be something new we'd add (essentially adding an extra line of code like we're already doing with the @changed parameter). 

As for creating a contextual component and using it to wrap the radio button and then creating a function that helps to reset that input value --> I think this would present the same scenario as editing existing components. We would still have to pass in an extra parameter to the component, and I think creating an entirely knew contextual component just to handle what could be handled in the template by a mut helper might over-complicate things / cause confusion for other developers trying to decipher the code.

I know that we were trying to pull things out of `edit.hbs` and really simplify that template but I think using the mut helper with the `@changed` parameter is the simplest way to go about it. 

Let me know if you agree/disagree with any of this! ^^ Happy to chat :)

Reply:
> As for creating a contextual component and using it to wrap the radio button and then creating a function that helps to reset that input value --> I think this would present the same scenario as editing existing components. We would still have to pass in an extra parameter to the component, and I think creating an entirely knew contextual component just to handle what could be handled in the template by a mut helper might over-complicate things / cause confusion for other developers trying to decipher the code.

Great! This sounds like very sound reasoning. Thanks for looking deeper :)

Reply:
Thanks for challenging me to think about all the options!